### PR TITLE
[sdl2] Update to 2.0.12 version

### PR DESCRIPTION
**Pull request by RT2Code**

### CODE REVIEW

**Comment by Cheney-W** [comment]
@dan-shaw 
Since the version of sdl2 on github: https://github.com/SDL-mirror/SDL has not been updated to the latest version, do we need to download the source from the sdl2 official website: https://www.libsdl.org/download-2.0.php?

Related issue: https://github.com/microsoft/vcpkg/issues/10429

**Comment by dan-shaw** [comment]
@Cheney-W I think downloading from the official website is fine. The release on the GitHub repo seems to be 2 versions behind.

File: ports/sdl2/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: sdl2
-Version: 2.0.10-3
+Version: 2.0.12
 Homepage: https://github.com/SDL-Mirror/SDL
```
Comment (Cheney-W) [inline ports/sdl2/CONTROL]: Could you please modify the homepage as `https://www.libsdl.org/download-2.0.php`?

File: ports/sdl2/portfile.cmake [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-    REPO SDL-Mirror/SDL
-    REF release-2.0.10
-    SHA512 c5fe59eed7ba9c6a82cceaf513623480793727fceec84b01d819e7cbefc8229a84be93067d7539f12d5811c49d3d54fd407272786aef3e419f439d0105c34b21
-    HEAD_REF master
+    ARCHIVE ${ARCHIVE}
     PATCHES
         export-symbols-only-in-shared-build.patch
-        fix-cmake-include-dir.patch
         enable-winrt-cmake.patch
-        fix-arm64-headers.patch
```
Comment (Cheney-W) [inline ports/sdl2/portfile.cmake]: Could you please delete patch files that no longer use?

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: ports/sdl2/portfile.cmake (hunk 2 above)
Comment (Cheney-W) [reply ports/sdl2/portfile.cmake]: I think you should delete the fix-cmake-include-dir.patch instead of enable-winrt-cmake.patch.

Comment (RT222) [reply ports/sdl2/portfile.cmake]: Sorry about that. I fixed it and removed two more unused patches. It should be fine now.

**Comment by Cheney-W** [comment]
All features passed with following triplets:
`x86-windows (Pass)`
`x64-windows (Pass)`
`x64-windows-static (Pass)`

**Review by Cheney-W — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**